library/modules/EventManager.cpp

```diff
@@ -597,57 +630,71 @@ static void manageJobCompletedEvent(color_ostream& out) {
     }
 #endif
 
-    for (auto &prevJob : prevJobs) {
-        //if it happened within a tick, must have been cancelled by the user or a plugin: not completed
-        if ( tick1 <= tick0 )
-            continue;
-
-        if ( nowJobs.find(prevJob.first) != nowJobs.end() ) {
-            //could have just finished if it's a repeat job
-            df::job& job0 = *prevJob.second;
-            if ( !job0.flags.bits.repeat )
-                continue;
-            df::job& job1 = *nowJobs[prevJob.first];
-            if ( job0.completion_timer != 0 )
-                continue;
-            if ( job1.completion_timer != -1 )
-                continue;
-
-            //still false positive if cancelled at EXACTLY the right time, but experiments show this doesn't happen
-            for (auto &[_,handle] : copy) {
-                DEBUG(log,out).print("calling handler for repeated job completed event\n");
-                run_handler(out, EventType::JOB_COMPLETED, handle, (void*) &job0);
+    // if it happened within a tick, must have been cancelled by the user or a plugin: not completed
+    // note observation: delta 0 tick events happens normally on load, save, paused (whether by user or game)
+    if (tick1 >= tick0) {
+        auto prevIt = prevJobs.begin();
+        auto nowIt = nowJobs.begin();
+        while (prevIt != prevJobs.end() && nowIt != nowJobs.end()) {
```

Comment:
I think you still need to process the remaining `prevJobs`, even after `nowJobs` has reached the end. Alternatively, you can always iterate to the end of `prevJobs`, and only advance `nowJobs` if it isn't at the end already.